pytest_plt/plugin.py

```diff
@@ -88,7 +90,19 @@ def dirname(self, _dirname):
         self._dirname = _dirname
 
     def get_filename(self, ext=""):
-        return "%s.%s" % (self.nodeid, ext)
+        filename = self.nodeid
+
+        # flat filestructure (replace folders with dots in filename)
+        filename = filename.replace(os.path.sep, '.')
+
+        # drop parts of filename matching the given regexs
```

Comment:
I think dropping arbitrary parts of the filename can result in collisions, whereas only dropping prefixes cannot result in collisions. So I think I would err on the side of only allowing prefixes to be stripped.

Reply:
Dropping prefixes can also result in collisions, I think it'd just be really uncommon. For example, if you drop the `nengo.tests` prefix and you have files in `nengo/tests/nengo/builder/tests/test_stuff.py` and `nengo/builder/tests/test_stuff.py`, that would result in a collision. But probably nobody is going to do that.

The thing I don't like about just prefixes is then it's impossible to get rid of things like `tests` folders throughout the project, which at least for our nengo projects is a common pattern and something that we could safely drop. If people only want to match prefixes, they can always start their regex with a carat, and we can advertise that (e.g. make it clear that it's much safer to use a carat and only drop prefixes, and that if using other regexes, you need to be aware of possible collisions). However, I don't like the hand-holding approach of not allowing something just because it could potentially be used incorrectly.

Reply:
> Dropping prefixes can also result in collisions, I think it'd just be really uncommon. For example, if you drop the `nengo.tests` prefix and you have files in `nengo/tests/nengo/builder/tests/test_stuff.py` and `nengo/builder/tests/test_stuff.py`, that would result in a collision. But probably nobody is going to do that.

Right, that's true, didn't think about that case. That makes sense then since the regex can have the carat, I think we should just make sure in documentation to note that using `filename_drop` can result in collisions, and we should also provide some example regexes or common things like removing `tests.`. I'll add that to #4.